ToDo/Program.cs

```diff
@@ -1,200 +1,191 @@
+        private static int _positionInTask = 0;
+        private static int _numberOfTask;
+        private static Menu _menu = new("Main Menu");
+
+        static void Main(string[] args)
+        {
+            _menu.AddNewItem("add new task", "a", ExecuteAddNewTask);
+            _menu.AddNewItem("add description", "e", ExecuteAddDescription);
+            _menu.AddNewItem("remove task", "x", ExecuteDeleteTask);
+            _menu.AddNewItem("mark task as done", "d", () => ExecuteStatusTask("done"));
+            _menu.AddNewItem("mark task as undone", "u", () => ExecuteStatusTask("undone"));
+            _menu.AddNewItem("move up", "w", () => ExecuteChangePosition(_positionInTask, "up"));
+            _menu.AddNewItem("move down", "s", () => ExecuteChangePosition(_positionInTask, "down"));
+
+            while (true)
+            {
+                Console.WriteLine(NavigationMenu());
+                //Console.WriteLine(_numberOfTasks);
+                Console.WriteLine(_positionInTask);
+
+                ShowTaskList();
+                _numberOfTask = GetNumberOfTask();
+
+                var userChoice = Console.ReadKey();
+                string key = userChoice.Key.ToString().ToLower();
+                Console.WriteLine(key);
+                switch (userChoice.Key)
+                {
+                    case ConsoleKey.A:
+                    case ConsoleKey.E:
+                    case ConsoleKey.D:
+                    case ConsoleKey.U:
+                    case ConsoleKey.X:
+                    case ConsoleKey.W:
+                    case ConsoleKey.S:
+                        _menu.ExecuteItem(key);
+                        break;
+                    case ConsoleKey.Escape:
+                        return;
+                    default:
+                        Console.Write("Wrong button, try again: ");
+                        break;
+                }
+                Console.Clear();
+            }
+        }
+
+        public static void ExecuteAddNewTask()
+        {
+            Console.Write("\nEnter name of task: ");
+            string taskTitle = Console.ReadLine();
+            while (!IsValid(taskTitle))
+            {
+                Console.Write("The name of task can not be empty!\nEnter name of task: ");
+
+                taskTitle = Console.ReadLine();
+            }
+            Task newTask = new(taskTitle);
+            _listOfTasks.AddToList(newTask);
+        }
+
+        public static void ExecuteAddDescription()
+        {
+            Task task = _listOfTasks.GetTaskList()[_positionInTask];
+            Console.Write($"\nEnter desription for {task.Title}: ");
+            string userInput = Console.ReadLine();
+            task.AddDescription(userInput);
+        }
+
+        public static void ExecuteStatusTask(string status)
+        {
+            if (status == "done")
+            {
+                _listOfTasks.GetTaskList()[_positionInTask].MarkAsDone();

```

Comment:
A couple of issues here:
- No check if index is valid
- Getting item by index should be encapsulated into the TaskList itself
- GetTaskList is not a valid name, because you are calling a method on TaskList type